lib/diagnostic.rb

```diff
@@ -7,36 +7,36 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max { |a, b| a.length <=> b.length}
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.inject(0) { |sum, x| sum + x }
```

Comment:
This didn't quite work, close though. Here's a quick way to do it:

`list.reduce(&:+)`